contracts/PortfolioLib.sol

```diff
@@ -192,92 +197,101 @@ function changePositionLiquidity(
+    uint256 totalLiquidity = self.liquidity;
+    if (totalLiquidity == 0) totalLiquidity = INIT_LIQUIDITY; // use 1E18 of liquidity
 
     (uint256 amountAssetWad, uint256 amountQuoteWad) =
-        self.getVirtualPoolReservesPerLiquidityInWad();
-    uint256 liquidity0 = deltaAsset.divWadDown(amountAssetWad); // L_0 = X / (X / L)
-    uint256 liquidity1 = deltaQuote.divWadDown(amountQuoteWad); // L_1 = Y / (Y / L)
+        self.getVirtualReservesWad();
+    uint256 liquidity0 = deltaAsset.mulDivDown(totalLiquidity, amountAssetWad); // L_0 = dX * L / X
+    uint256 liquidity1 = deltaQuote.mulDivDown(totalLiquidity, amountQuoteWad); // L_1 = dY * L / Y
     deltaLiquidity = AssemblyLib.min(liquidity0, liquidity1).safeCastTo128();
 }
 
 /**
  * @dev Rounds positive deltas up. Rounds negative deltas down.
+ * @return deltaAsset Real `asset` tokens underlying `deltaLiquidity`, denominated in WAD.
+ * @return deltaQuote Real `quote` tokens underlying `deltaLiquidity`, denominated in WAD.
  */
 function getPoolLiquidityDeltas(
     PortfolioPool memory self,
     int128 deltaLiquidity
 ) pure returns (uint128 deltaAsset, uint128 deltaQuote) {
     if (deltaLiquidity == 0) return (deltaAsset, deltaQuote);
 
+    uint256 delta;
+    uint256 totalLiquidity = self.liquidity;
     (uint256 amountAssetWad, uint256 amountQuoteWad) =
-        self.getVirtualPoolReservesPerLiquidityInWad();
-    uint256 scaleDownFactorAsset = AssemblyLib.computeScalar(
-        self.pair.decimalsAsset
-    ) * FixedPointMathLib.WAD;
-    uint256 scaleDownFactorQuote = AssemblyLib.computeScalar(
-        self.pair.decimalsQuote
-    ) * FixedPointMathLib.WAD;
+        self.getVirtualReservesWad();
+
+    // Pre-allocate pools initialize reserves to 1E18 of liquidity to
+    // compute the first allocate quantities.
+    if (self.liquidity == 0) {
+        totalLiquidity = 1e18;
+    }
 
-    uint256 delta;
     if (deltaLiquidity > 0) {
+        // If allocating liquidity, round token amounts up.
         delta = uint128(deltaLiquidity);
         deltaAsset =
-            amountAssetWad.mulDivUp(delta, scaleDownFactorAsset).safeCastTo128();
+            delta.mulDivUp(amountAssetWad, totalLiquidity).safeCastTo128();
         deltaQuote =
-            amountQuoteWad.mulDivUp(delta, scaleDownFactorQuote).safeCastTo128();
+            delta.mulDivUp(amountQuoteWad, totalLiquidity).safeCastTo128();
     } else {
+        // If deallocating liquidity, round token amounts down.
         delta = uint128(-deltaLiquidity);
-        deltaAsset = amountAssetWad.mulDivDown(delta, scaleDownFactorAsset)
-            .safeCastTo128();
-        deltaQuote = amountQuoteWad.mulDivDown(delta, scaleDownFactorQuote)
-            .safeCastTo128();
+        deltaAsset =
+            delta.mulDivDown(amountAssetWad, totalLiquidity).safeCastTo128();
+        deltaQuote =
+            delta.mulDivDown(amountQuoteWad, totalLiquidity).safeCastTo128();
     }
```

Comment:
No longer scales the delta's down to native token decimal units. Also computes the deltas using the full reserves and full liquidity.